Rework OSDK intros + Golang examples (#28372)
openshift/openshift-docs: modules/osdk-golang-manager.adoc

// Module included in the following assemblies:
//
// * operators/operator_sdk/osdk-golang-tutorial.adoc

[id="osdk-golang-manager_{context}"]
= About the Manager

> The main program for the Operator is the `main.go` file, which initializes and runs the link:https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/manager#Manager[Manager]. The Manager automatically registers the Scheme for all custom resource (CR) API definitions, and sets up and runs controllers and webhooks.

Reviewer: ```suggestion
The main program for the Operator is the `main.go` file, which initializes and runs the link:https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/manager#Manager[Manager]. The Manager automatically registers the Scheme for all custom resource (CR) API definitions and sets up and runs controllers and webhooks.
```

The Manager can restrict the namespace that all controllers watch for resources:

[source,go]
----
mgr, err := ctrl.NewManager(cfg, manager.Options{Namespace: namespace})
----

> By default, this is the namespace where the Operator runs. To watch all namespaces, you can leave the `namespace` option empty:

Reviewer: ```suggestion
By default, the Manager watches the namespace where the Operator runs. To watch all namespaces, you can leave the `namespace` option empty:
```

[source,go]
----
mgr, err := ctrl.NewManager(cfg, manager.Options{Namespace: ""})
----

> It is also possible to use the link:https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/cache#MultiNamespacedCacheBuilder[`MultiNamespacedCacheBuilder`] function to watch a specific set of namespaces:

Reviewer: ```suggestion
You can also use the link:https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/cache#MultiNamespacedCacheBuilder[`MultiNamespacedCacheBuilder`] function to watch a specific set of namespaces:
```

[source,go]
----
var namespaces []string <1>
mgr, err := ctrl.NewManager(cfg, manager.Options{ <2>
   NewCache: cache.MultiNamespacedCacheBuilder(namespaces),
})
----

> <1> List of namespaces.

Reviewer: Is this just a comma-separated list, or does the variable expect some other syntax?

> <2> Creates a new `Cmd` to provide shared dependencies and start components.

Reviewer: ```suggestion
<2> Creates a `Cmd` object to provide shared dependencies and start components.
```
